src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12611,11 +12477,11 @@ private AnonymousObjectCreationExpressionSyntax ParseAnonymousTypeExpression()
             var openBrace = this.EatToken(SyntaxKind.OpenBraceToken);
             var expressions = _pool.AllocateSeparated<AnonymousObjectMemberDeclaratorSyntax>();
             this.ParseAnonymousTypeMemberInitializers(ref openBrace, ref expressions);
-            var closeBrace = this.EatToken(SyntaxKind.CloseBraceToken);
-            var result = _syntaxFactory.AnonymousObjectCreationExpression(@new, openBrace, expressions, closeBrace);
-            _pool.Free(expressions);
```

Comment:
an example of needing to break things up over multiple statements to free things.  Both this pattern and thr try/finally pattern go away with the simple ToListAndFree pattern that already existed.